torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3866,33 +3862,100 @@ def sample_inputs_conv1d(op_info, device, dtype, requires_grad, **kwargs):
+
+    # error inputs for negative strides
     yield ErrorInput(
-        SampleInput(input, kwargs={"weight": weight, "groups": groups}),
-        error_regex="non-positive groups is not supported"
-    )
+        SampleInput(make_arg((1, 1, 4)), args=(make_arg((1, 2, 2)), make_arg((1,))),
+                    kwargs={'stride': (-1,)}), error_regex="non-positive stride is not supported")
+
+    # error inputs for negative padding
+    yield ErrorInput(
+        SampleInput(make_arg((1, 1, 4)), args=(make_arg((1, 2, 2)), make_arg((1,))),
+                    kwargs={'padding': (-1,)}), error_regex="negative padding is not supported")
+
+    # error inputs for negative dilation
+    yield ErrorInput(
+        SampleInput(make_arg((1, 1, 4)), args=(make_arg((1, 1, 2)), make_arg((1,))),
+                    kwargs={'dilation': (-1,)}), error_regex="dilation should be greater than zero")
+
+    # FIXME: https://github.com/pytorch/pytorch/issues/85656
+    # error inputs for bias shape not equal to the output channels
+    # yield ErrorInput(SampleInput(make_arg((1, 1, 4)), args=(make_arg((1, 1, 3)), make_arg((2,)))),
+    #                  error_regex="expected bias to be 1-dimensional with 1 elements")
+
+    # error inputs for input.ndim != weight.ndim
+    yield ErrorInput(SampleInput(make_arg((1, 1, 4)), args=(make_arg((1, 2)), make_arg((1,)))),
+                     error_regex="weight should have at least three dimensions")
+
+    # error inputs for the weight[0] are less than the number of groups
+    yield ErrorInput(
+        SampleInput(make_arg((2, 2, 4)), args=(make_arg((2, 2, 2)), make_arg((2,))),
+                    kwargs={'padding': 'same', 'groups': 3}), error_regex="expected weight to be at least 3 at dimension 0")
+
+    # error inputs for the weight[0] are less than the number of groups
+    yield ErrorInput(
+        SampleInput(make_arg((2, 2, 4)), args=(make_arg((2, 2, 2)), make_arg((2,))),
+                    kwargs={'groups': 3}), error_regex="expected weight to be at least 3 at dimension 0")
+
+    # error inputs for invalid groups
+    yield ErrorInput(
+        SampleInput(make_arg((2, 2, 4)), args=(make_arg((2, 2, 2)), make_arg((2,))),
+                    kwargs={'padding': 'same', 'groups': -1}), error_regex="non-positive groups is not supported")
+
+    # error inputs for invalid groups
+    yield ErrorInput(
+        SampleInput(make_arg((2, 2, 4)), args=(make_arg((2, 2, 2)), make_arg((2,))),
+                    kwargs={'padding': 'same', 'groups': 0}), error_regex="non-positive groups is not supported")
 
 
 def error_inputs_conv2d(opinfo, device, **kwargs):
-    weight = torch.randint(high=10, size=(3, 2, 3, 3), device=device)
-    input = torch.randint(high=10, size=(2, 4, 4), device=device)
-    bias = torch.rand((3,), dtype=torch.float32, device=device)
-    yield ErrorInput(SampleInput(input, args=(weight, bias)), error_regex="should be the same")
```

Comment:
Why did you remove this sample (and the one below)? Where is this tested?

Reply:
Added this and the below case, using `make_arg` in my latest commit https://github.com/pytorch/pytorch/pull/113885/commits/a358f6c06bbcd57ddc12941aeceb0a752a44f3b2

Reply:
do it in all the conv operators please

Reply:
@lezcano, `conv1d`, and `conv3d` does not throw error, unlike `conv2d`. Please see:

**For conv1d:**
```python
>>> import torch
>>> input = torch.randint(high=10, size=(1, 1, 4))
>>> weight = torch.randint(high=10, size=(1, 1, 2))
>>> bias = torch.rand(1, dtype=torch.float64)
>>> torch.nn.functional.conv1d(input, weight, bias)
tensor([[[86, 97, 90]]])
>>> bias = torch.rand(1, dtype=torch.complex128)
>>> torch.nn.functional.conv1d(input, weight, bias)
<stdin>:1: UserWarning: Casting complex values to real discards the imaginary part (Triggered internally at /home/khushi/Documents/hello/pytorch/aten/src/ATen/native/Copy.cpp:301.)
tensor([[[86, 97, 90]]])
```
**For conv3d:**
```python
>>> input = torch.randint(high=10, size=(1, 1, 4, 4, 4))
>>> weight = torch.randint(high=10, size=(1, 1, 2, 2, 2))
>>> bias = torch.rand(1, dtype=torch.float64)
>>> torch.nn.functional.conv3d(input, weight, bias)
tensor([[[[[196, 192, 212],
           [155, 184, 136],
           [123, 152, 111]],

          [[161, 230, 240],
           [189, 240, 236],
           [214, 137, 105]],

          [[137, 196, 232],
           [127, 187, 293],
           [118, 206, 196]]]]])
>>> bias = torch.rand(1, dtype=torch.complex128)
>>> torch.nn.functional.conv3d(input, weight, bias)
tensor([[[[[196, 192, 212],
           [155, 184, 136],
           [123, 152, 111]],

          [[161, 230, 240],
           [189, 240, 236],
           [214, 137, 105]],

          [[137, 196, 232],
           [127, 187, 293],
           [118, 206, 196]]]]])
```
I think we should fix this, maybe in a different PR. I'll work on this. wdyt? 

Thanks!

Reply:
Sure, different PR SGTM